src/clients/discord.ts

```diff
@@ -60,13 +60,42 @@ function registerDiscord(): void
         const messageToSend: string = `${ discordHeader }${ finalMessage }`;
 
         //Create a key-value pair that will be logged as a partially complete fused object. When we find the other piece on the twitch side, it will also be mapped in our collection.
-        const listNode: linkedListNode = bridge.messageLinkdListInterface.addNode(new conjoinedMsg(m));
+        const listNode: linkedListNode = bridge.messageLinkedListInterface.addNode(new conjoinedMsg(m));
         bridge.twitchMessageSearchCache[messageToSend] = listNode;
         bridge.discordTwitchCacheMap.set(m, listNode);
+        bridge.discordTwitchCacheMap.set(m.id, listNode);
 
-        //I'm only grabbing the first index here... this will need to change if we scale up.
-        bridge.twitchClient!.say(configFile.T2D_CHANNELS[0], messageToSend).then(undefined, genericPromiseError);
+        // if the message is a reply
+        if(m.type === 'REPLY')
+        {
+            // fetchReference will fetch the message that was replied to.
+            // Somewhere here is going wrong, we want to get the
+            // replied message's (the fetchReference)'s Twitch message object
+            // although the ID is alwasys weirdly wrong?
+            const fetchedReplyMessage: Message<boolean> = await m.fetchReference();
+            const replyNode: linkedListNode | undefined = bridge.discordTwitchCacheMap.get(fetchedReplyMessage.id);
+            
+            console.log(replyNode);
+            console.log(replyNode?.data.twitchArray[0]);
+
+            const twitchMsgData = replyNode?.data.twitchArray[0];
+            console.log(twitchMsgData?.userState['id'] == replyNode?.data.twitchArray[0].userState['id']);
+            // Look at Twitch docs, this is a raw statement cause TMI is bad (:.
+            // This just replies to our message using a Twitch ID.
+            // THIS IS ALWAYS BEHIND CAUSE TMI IS BAD.
+            // THIS WILL NOT WORKY YET-Y
+            if(twitchMsgData?.userState.id && !bridge.lastUserStateMsg.userState.botUserStateId)
+                bridge.twitchClient?.raw(`@reply-parent-msg-id=${ twitchMsgData!.userState.id } PRIVMSG ${ configFile.T2D_CHANNELS[0] } :${ messageToSend }`);
```

Comment:
Shame tmi.js is bugged, this is a really fascinating piece of functionality!

Dang, fascinating to see how tmi works behind the scenes through this string alone

Reply:
(: To be honest, making a custom wrapper for the Twitch API wouldn't be that hard; the raw data we send is very simple. New project idea to abandon in a month! Prepare for "phan" Twitch wrapper! /hj